frodo.js

```diff
@@ -1,135 +1,129 @@
-/**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.